src/solvers/smt2/smt2_parser.cpp

```diff
@@ -975,6 +975,38 @@ typet smt2_parsert::sort()
         return nil_typet();
       }
     }
+    else if(buffer == "Array")
+    {
+      // this gets two sorts as arguments, domain and range
+      auto domain = sort();
+      auto range = sort();
+
+      // eat the ')'
+      if(next_token() != CLOSE)
+      {
+        error() << "expected ')' at end of Array sort" << eom;
+        return nil_typet();
+      }
+
+      // we can turn arrays that map a bitvector type to something else
+      // into our 'array_typet'
+      if(domain.id() == ID_unsignedbv)
+      {
+        const auto size = to_unsignedbv_type(domain).largest_expr();
+        return array_typet(range, size);
+      }
+      else if(domain.id() == ID_signedbv)
```

Comment:
What would generate these?  Seeing as SMT-LIB doesn't have a signed bit-vector sort and the BitVec is being treated as unsigned?